cmd/batch-expire.go

```diff
@@ -0,0 +1,479 @@
+type BatchExpireNotification struct {
+	Endpoint string `yaml:"endpoint" json:"endpoint"`
+	Token    string `yaml:"token" json:"token"`
+}
+
+// BatchJobExpireFlags various configurations for replication job definition currently includes
+// - filter
+// - notify
+// - retry
+type BatchJobExpireFlags struct {
+	Filter BatchExpireFilter       `yaml:"filter" json:"filter"`
+	Notify BatchExpireNotification `yaml:"notify" json:"notify"`
+	Retry  BatchExpireRetry        `yaml:"retry" json:"retry"`
+}
+
+// BatchJobExpireV1 v1 of batch key rotation job
+type BatchJobExpireV1 struct {
+	APIVersion string              `yaml:"apiVersion" json:"apiVersion"`
+	Flags      BatchJobExpireFlags `yaml:"flags" json:"flags"`
+	Bucket     string              `yaml:"bucket" json:"bucket"`
+	Prefix     string              `yaml:"prefix" json:"prefix"`
+}
+
+// Notify notifies notification endpoint if configured regarding job failure or success.
+func (r BatchJobExpireV1) Notify(ctx context.Context, body io.Reader) error {
+	if r.Flags.Notify.Endpoint == "" {
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, r.Flags.Notify.Endpoint, body)
+	if err != nil {
+		return err
+	}
+
+	if r.Flags.Notify.Token != "" {
+		req.Header.Set("Authorization", r.Flags.Notify.Token)
+	}
+
+	clnt := http.Client{Transport: getRemoteInstanceTransport}
+	resp, err := clnt.Do(req)
+	if err != nil {
+		return err
+	}
+
+	xhttp.DrainBody(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		return errors.New(resp.Status)
+	}
+
+	return nil
+}
+
+// Expire rotates encryption key of an object
+func (r *BatchJobExpireV1) Expire(ctx context.Context, api ObjectLayer, objInfo ObjectInfo) error {
+	srcBucket := r.Bucket
+	srcObject := objInfo.Name
+
+	_, err := api.DeleteObject(ctx, srcBucket, srcObject, ObjectOptions{
+		DeletePrefix: true,
+	})
+	return err
+}
+
+const (
+	batchExpireName                 = "batch-expire.bin"
+	batchExpireFormat               = 1
+	batchExpireVersionV1            = 1
+	batchExpireVersion              = batchExpireVersionV1
+	batchExpireAPIVersion           = "v1"
+	batchExpireJobDefaultRetries    = 3
+	batchExpireJobDefaultRetryDelay = 250 * time.Millisecond
+)
+
+// Start the batch key rottion job, resumes if there was a pending job via "job.ID"
+func (r *BatchJobExpireV1) Start(ctx context.Context, api ObjectLayer, job BatchJobRequest) error {
+	ri := &batchJobInfo{
+		JobID:     job.ID,
+		JobType:   string(job.Type()),
+		StartTime: job.Started,
+	}
+	if err := ri.load(ctx, api, job); err != nil {
+		return err
+	}
+
+	globalBatchJobsMetrics.save(job.ID, ri)
+	lastObject := ri.Object
+
+	delay := job.Expire.Flags.Retry.Delay
+	if delay == 0 {
+		delay = batchExpireJobDefaultRetryDelay
+	}
+	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	skip := func(info FileInfo) (ok bool) {
```

Comment:
```suggestion
	filter := func(info FileInfo) (ok bool) {
```
`opts.WalkFitler` takes a function which returns true if the entry should be returned as part of the Walk calls results. `skip` gives the opposite impression/idea. This is a minor change but will help future readers understand the intent better.